torch/_subclasses/fake_tensor.py

```diff
@@ -965,6 +977,163 @@ def should_allow_numbers_as_tensors(func: OpOverload):
     )
 
 
+_DISPATCH_CACHE = {}
+
+
+class _Boolean(NamedTuple):
+    value: bool
+
+
+_True = _Boolean(True)
+_False = _Boolean(False)
+
+
+class FakeTensorMetadata(NamedTuple):
+    size: Tuple[int, ...]
+    stride: Tuple[int, ...]
+    dtype: torch.dtype
+    device: torch.device
+    requires_grad: bool
+    is_view: bool
+    is_inference: bool
+    constant: Any
+
+
+def _dispatch_should_fallback(op: Any) -> bool:
+    if op == torch.ops.prim.device.default:
+        return True
+
+    assert isinstance(op, OpOverload)
+    schema = op._schema
+
+    # Return true if there's any alias information in any of the
+    # return values.
+    return len(schema.returns) > 0 and any(
+        r.alias_info is not None for r in schema.returns
```

Comment:
This adds a dependency on the schema being correct which did not exist previously. Particularly for internal operators this might be wrong. I would prefer if we detected aliasing at runtime when running the operator.